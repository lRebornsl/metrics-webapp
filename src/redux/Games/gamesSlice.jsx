import { createAsyncThunk, createReducer } from "@reduxjs/toolkit";
import { URL } from "../../utils/constant";
import axios from "axios";

const GET_GAMES_BY_CATEGORY = "gamesStore/games/GET_GAMES"

export const getGamesByCategory = createAsyncThunk(GET_GAMES_BY_CATEGORY, async (category) => {
  const response = await axios.get(URL, {
    params: {
      category: category
    },
    headers: {
    'X-RapidAPI-Key': '036dcaeac8msh963d8e43a0bf8b0p18b172jsnf6ae43950a75',
    'X-RapidAPI-Host': 'mmo-games.p.rapidapi.com'
    }
  });
  return response.data;
})

const initialState = {
  games: [],
};

const gamesSlice = createReducer(initialState, builder => {
  builder
    .addCase(getGamesByCategory.fulfilled, (state, { payload }) => {
      return{
        ...state,
        games: [...payload],
      }
    })
})

export default gamesSlice;