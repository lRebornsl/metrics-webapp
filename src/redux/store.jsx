import { configureStore } from '@reduxjs/toolkit';
import categoriesSlice from './Categories/categoriesSlice';

const store = configureStore({
  reducer: {
    categoriesSlice,
  },
});

export default store;