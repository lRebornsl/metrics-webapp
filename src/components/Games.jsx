import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { useLocation } from "react-router-dom";
import { getGamesByCategory } from "../redux/Games/gamesSlice";

const Games = () => {
  const location = useLocation();
  const categoryName = location.state;
  const dispatch = useDispatch();
  const games = useSelector(state => state.gamesSlice.games)

  useEffect(() => {
    dispatch(getGamesByCategory(categoryName))
  }, [dispatch, categoryName]);

  return(
    <div>
      {games.map(game => (
        <div className="text-white" key={game.id}>
          <a href={game.game_url}><img className="w-full border-y-2 border-white" src={game.thumbnail} alt="Image" /></a>
          <div className="flex justify-between px-4 py-2 text-xl font-bold bg-queen border-b-2 border-white">
            <h2>{game.title}</h2>
            <h2>{game.release_date}</h2>
          </div>
          <p className="px-4 py-2 text-justify bg-united">{game.short_description}</p>
        </div>
      ))}
    </div>
  )
}

export default Games;