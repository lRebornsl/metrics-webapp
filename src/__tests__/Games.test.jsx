import { render, waitFor } from "@testing-library/react";
import { Provider } from "react-redux";
import { beforeEach, describe, expect, it } from "vitest";
import store from "../redux/store";
import { MemoryRouter } from "react-router-dom";
import Games from "../components/Games";

describe("Games tests", () => {

  beforeEach(() => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Games />
        </MemoryRouter>
      </Provider>
    );
  });

  it("Games renders correctly", async () => {
    await waitFor(() => expect(Games).toMatchSnapshot());
  });

  it("Images are rendered correctly", async () => {
    const gameImages = document.querySelectorAll("img");
    expect(gameImages).toMatchSnapshot();
  });
})