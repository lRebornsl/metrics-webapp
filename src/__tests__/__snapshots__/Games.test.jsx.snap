// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Games tests > Games renders correctly 1`] = `[Function]`;

exports[`Games tests > Images are rendered correctly 1`] = `NodeList []`;
